### ISSUE TITLE
App crashes

### ISSUE THREAD

**Issue by 821938089** [fast4x/RiMusic#1052]
### Steps to reproduce the bug

Crash 1: A song fails to play after a while
Crash 2: When opening the playlist

### Expected behavior

Not Crash

### Actual behavior

Crashed

### Logs

Crash:1
```
java.lang.IllegalStateException
 at a9.n1.I0(SourceFile:1)
 at u3.y.m(SourceFile:1)
 at y8.z.u(SourceFile:30)
 at y8.z.l(SourceFile:13)
 at ra.r.q(SourceFile:157)
 at ra.d1.b(SourceFile:177)
 at ra.c1.u(SourceFile:1)
 at y9.a.k(SourceFile:1)
 at oa.j0.run(SourceFile:1)
 at l.j.run(SourceFile:14)
 at ua.i.run(SourceFile:1)
 at ua.a.run(SourceFile:92)
 Suppressed: ta.h: [v1{Cancelling}@4e307eb, Dispatchers.IO]
```
```
java.lang.IllegalStateException
 at androidx.media3.common.util.Assertions.checkState(Assertions.java:85)
 at androidx.media3.datasource.cache.SimpleCache.isCached(SimpleCache.java:459)
 at it.vfsfitvnm.vimusic.service.PlayerService$isDownloadedState$1.invokeSuspend(PlayerService.kt:334)
 at it.vfsfitvnm.vimusic.service.PlayerService$isDownloadedState$1.invoke(Unknown Source:8)
 at it.vfsfitvnm.vimusic.service.PlayerService$isDownloadedState$1.invoke(Unknown Source:4)
 at kotlinx.coroutines.flow.FlowKt__MergeKt$flatMapMerge$$inlined$map$1$2.emit(Emitters.kt:223)
 at kotlinx.coroutines.flow.StateFlowImpl.collect(StateFlow.kt:396)
 at kotlinx.coroutines.flow.StateFlowImpl$collect$1.invokeSuspend(Unknown Source:15)
 at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
 at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
 at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
 at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
 at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
 at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
 at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
 at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
 Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@ac15b96, Dispatchers.IO]
```

Crash2:
```
java.lang.IndexOutOfBoundsException: Index -1, size 1
 at y.c1.b(SourceFile:27)
 at y.c1.c(SourceFile:1)
 at xb.e.E(SourceFile:5)
 at w.p.b(SourceFile:1)
 at w.v.a(SourceFile:7)
 at w.w.l(SourceFile:759)
 at v.v.b(SourceFile:24)
 at v.v.l(SourceFile:219)
 at l1.d0.a(SourceFile:74)
 at n1.x.a(SourceFile:38)
 at s.m.a(SourceFile:9)
 at s.m.f(SourceFile:14)
 at l1.x.h(SourceFile:8)
 at n1.c0.a(SourceFile:11)
 at s.m.a(SourceFile:39)
 at s.m.f(SourceFile:27)
 at l1.x.h(SourceFile:8)
 at n1.c0.a(SourceFile:11)
 at y0.q0.h(SourceFile:1)
 at n1.c0.a(SourceFile:11)
 at r.l0.a(SourceFile:42)
 at r.l0.d(SourceFile:137)
 at g0.l.d(SourceFile:60)
 at q0.a0.a(SourceFile:48)
 at q0.b0.c(SourceFile:153)
 at n1.o1.a(SourceFile:1)
 at n1.l0.n0(SourceFile:107)
 at n1.l0.a(SourceFile:96)
 at v.t.a(SourceFile:219)
 at n1.x.a(SourceFile:38)
 at r.l0.a(SourceFile:42)
 at r.l0.d(SourceFile:137)
 at g0.l.d(SourceFile:60)
 at q0.a0.a(SourceFile:48)
 at q0.b0.c(SourceFile:153)
 at n1.o1.a(SourceFile:1)
 at n1.l0.n0(SourceFile:107)
 at n1.l0.a(SourceFile:96)
 at v.x0.a(SourceFile:472)
 at n1.x.a(SourceFile:38)
 at r.l0.a(SourceFile:42)
 at r.l0.d(SourceFile:137)
 at g0.l.d(SourceFile:60)
 at q0.a0.a(SourceFile:48)
 at q0.b0.c(SourceFile:153)
 at n1.o1.a(SourceFile:1)
 at n1.l0.n0(SourceFile:107)
 at n1.l0.a(SourceFile:96)
 at v.t.a(SourceFile:219)
 at n1.x.a(SourceFile:38)
 at v.e0.h(SourceFile:99)
 at n1.c0.a(SourceFile:11)
 at v.r0.h(SourceFile:1)
 at n1.c0.a(SourceFile:11)
 at r.l0.a(SourceFile:42)
 at r.l0.d(SourceFile:137)
 at g0.l.d(SourceFile:60)
 at q0.a0.a(SourceFile:48)
 at q0.b0.c(SourceFile:153)
 at n1.o1.a(SourceFile:1)
 at n1.l0.n0(SourceFile:107)
 at n1.s0.c(SourceFile:55)
 at n1.s0.k(SourceFile:124)
 at n1.s0.h(SourceFile:77)
 at androidx.compose.ui.platform.AndroidComposeView.u(SourceFile:34)
 at n1.k1.a(SourceFile:4)
 at androidx.compose.ui.platform.AndroidComposeView.dispatchDraw(SourceFile:14)
 at android.view.View.draw(View.java:24648)
 at android.view.View.updateDisplayListIfDirty(View.java:23476)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:726)
 at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:735)
 at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:845)
 at android.view.ViewRootImpl.draw(ViewRootImpl.java:5533)
 at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:5224)
 at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:4352)
 at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2933)
 at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10224)
 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1626)
 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1635)
 at android.view.Choreographer.doCallbacks(Choreographer.java:1170)
 at android.view.Choreographer.doFrame(Choreographer.java:1074)
 at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1599)
 at android.os.Handler.handleCallback(Handler.java:958)
 at android.os.Handler.dispatchMessage(Handler.java:99)
 at android.os.Looper.loopOnce(Looper.java:224)
 at android.os.Looper.loop(Looper.java:318)
 at android.app.ActivityThread.main(ActivityThread.java:8759)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:561)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
```
```
java.lang.IndexOutOfBoundsException: Index -1, size 1
 at androidx.compose.foundation.lazy.layout.MutableIntervalList.checkIndexBounds(IntervalList.kt:183)
 at androidx.compose.foundation.lazy.layout.MutableIntervalList.get(IntervalList.kt:166)
 at androidx.compose.foundation.lazy.layout.LazyLayoutIntervalContent.getKey(LazyLayoutIntervalContent.kt:86)
 at androidx.compose.foundation.lazy.LazyListItemProviderImpl.getKey(LazyListItemProvider.kt:85)
 at androidx.compose.foundation.lazy.LazyListMeasuredItemProvider.getAndMeasure(LazyListMeasuredItemProvider.kt:46)
 at androidx.compose.foundation.lazy.LazyListMeasureKt.measureLazyList-5IMabDg(LazyListMeasure.kt:195)
 at androidx.compose.foundation.lazy.LazyListKt$rememberLazyListMeasurePolicy$1$1.invoke-0kLqBqw(LazyList.kt:313)
 at androidx.compose.foundation.lazy.LazyListKt$rememberLazyListMeasurePolicy$1$1.invoke(LazyList.kt:178)
 at androidx.compose.foundation.lazy.layout.LazyLayoutKt$LazyLayout$3$2$1.invoke-0kLqBqw(LazyLayout.kt:107)
 at androidx.compose.foundation.lazy.layout.LazyLayoutKt$LazyLayout$3$2$1.invoke(LazyLayout.kt:100)
 at androidx.compose.ui.layout.LayoutNodeSubcompositionsState$createMeasurePolicy$1.measure-3p2s80s(SubcomposeLayout.kt:709)
 at androidx.compose.ui.node.InnerNodeCoordinator.measure-BRTryo0(InnerNodeCoordinator.kt:126)
 at androidx.compose.foundation.AndroidOverscroll_androidKt$StretchOverscrollNonClippingLayer$2.invoke-3p2s80s(AndroidOverscroll.android.kt:584)
 at androidx.compose.foundation.AndroidOverscroll_androidKt$StretchOverscrollNonClippingLayer$2.invoke(AndroidOverscroll.android.kt:583)
 at androidx.compose.ui.layout.LayoutModifierImpl.measure-3p2s80s(LayoutModifier.kt:294)
 at androidx.compose.ui.node.LayoutModifierNodeCoordinator.measure-BRTryo0(LayoutModifierNodeCoordinator.kt:116)
 at androidx.compose.foundation.AndroidOverscroll_androidKt$StretchOverscrollNonClippingLayer$1.invoke-3p2s80s(AndroidOverscroll.android.kt:568)
 at androidx.compose.foundation.AndroidOverscroll_androidKt$StretchOverscrollNonClippingLayer$1.invoke(AndroidOverscroll.android.kt:567)
 at androidx.compose.ui.layout.LayoutModifierImpl.measure-3p2s80s(LayoutModifier.kt:294)
 at androidx.compose.ui.node.LayoutModifierNodeCoordinator.measure-BRTryo0(LayoutModifierNodeCoordinator.kt:116)
 at androidx.compose.ui.graphics.SimpleGraphicsLayerModifier.measure-3p2s80s(GraphicsLayerModifier.kt:646)
 at androidx.compose.ui.node.LayoutModifierNodeCoordinator.measure-BRTryo0(LayoutModifierNodeCoordinator.kt:116)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:252)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:251)
 at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:2303)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver$ObservedScopeMap.observe(SnapshotStateObserver.kt:500)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:256)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:133)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeMeasureSnapshotReads$ui_release(OwnerSnapshotObserver.kt:113)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:1617)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.access$performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:36)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.remeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:620)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.measure-BRTryo0(LayoutNodeLayoutDelegate.kt:596)
 at androidx.compose.foundation.layout.BoxMeasurePolicy.measure-3p2s80s(Box.kt:144)
 at androidx.compose.ui.node.InnerNodeCoordinator.measure-BRTryo0(InnerNodeCoordinator.kt:126)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:252)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:251)
 at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:2303)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver$ObservedScopeMap.observe(SnapshotStateObserver.kt:500)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:256)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:133)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeMeasureSnapshotReads$ui_release(OwnerSnapshotObserver.kt:113)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:1617)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.access$performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:36)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.remeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:620)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.measure-BRTryo0(LayoutNodeLayoutDelegate.kt:596)
 at androidx.compose.foundation.layout.RowColumnMeasurementHelper.measureWithoutPlacing-_EkL_-Y(RowColumnMeasurementHelper.kt:172)
 at androidx.compose.foundation.layout.RowColumnMeasurePolicy.measure-3p2s80s(RowColumnImpl.kt:72)
 at androidx.compose.ui.node.InnerNodeCoordinator.measure-BRTryo0(InnerNodeCoordinator.kt:126)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:252)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:251)
 at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:2303)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver$ObservedScopeMap.observe(SnapshotStateObserver.kt:500)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:256)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:133)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeMeasureSnapshotReads$ui_release(OwnerSnapshotObserver.kt:113)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:1617)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.access$performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:36)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.remeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:620)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.measure-BRTryo0(LayoutNodeLayoutDelegate.kt:596)
 at androidx.compose.foundation.layout.BoxMeasurePolicy.measure-3p2s80s(Box.kt:144)
 at androidx.compose.ui.node.InnerNodeCoordinator.measure-BRTryo0(InnerNodeCoordinator.kt:126)
 at androidx.compose.foundation.layout.FillNode.measure-3p2s80s(Size.kt:699)
 at androidx.compose.ui.node.LayoutModifierNodeCoordinator.measure-BRTryo0(LayoutModifierNodeCoordinator.kt:116)
 at androidx.compose.foundation.layout.OffsetPxNode.measure-3p2s80s(Offset.kt:245)
 at androidx.compose.ui.node.LayoutModifierNodeCoordinator.measure-BRTryo0(LayoutModifierNodeCoordinator.kt:116)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:252)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$performMeasureBlock$1.invoke(LayoutNodeLayoutDelegate.kt:251)
 at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:2303)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver$ObservedScopeMap.observe(SnapshotStateObserver.kt:500)
 at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:256)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:133)
 at androidx.compose.ui.node.OwnerSnapshotObserver.observeMeasureSnapshotReads$ui_release(OwnerSnapshotObserver.kt:113)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:1617)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate.access$performMeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:36)
 at androidx.compose.ui.node.LayoutNodeLayoutDelegate$MeasurePassDelegate.remeasure-BRTryo0(LayoutNodeLayoutDelegate.kt:620)
 at androidx.compose.ui.node.LayoutNode.remeasure-_Sx5XlM$ui_release(LayoutNode.kt:1145)
 at androidx.compose.ui.node.LayoutNode.remeasure-_Sx5XlM$ui_release$default(LayoutNode.kt:1136)
 at androidx.compose.ui.node.MeasureAndLayoutDelegate.doRemeasure-sdFAvZA(MeasureAndLayoutDelegate.kt:356)
 at androidx.compose.ui.node.MeasureAndLayoutDelegate.remeasureAndRelayoutIfNeeded(MeasureAndLayoutDelegate.kt:514)
 at androidx.compose.ui.node.MeasureAndLayoutDelegate.remeasureAndRelayoutIfNeeded$default(MeasureAndLayoutDelegate.kt:491)
 at androidx.compose.ui.node.MeasureAndLayoutDelegate.measureAndLayout(MeasureAndLayoutDelegate.kt:377)
 at androidx.compose.ui.platform.AndroidComposeView.measureAndLayout(AndroidComposeView.android.kt:971)
 at androidx.compose.ui.node.Owner$-CC.measureAndLayout$default(Owner.kt:228)
 at androidx.compose.ui.platform.AndroidComposeView.dispatchDraw(AndroidComposeView.android.kt:1224)
 at android.view.View.draw(View.java:24648)
 at android.view.View.updateDisplayListIfDirty(View.java:23476)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4607)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4580)
 at android.view.View.updateDisplayListIfDirty(View.java:23410)
 at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:726)
 at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:735)
 at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:845)
 at android.view.ViewRootImpl.draw(ViewRootImpl.java:5533)
 at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:5224)
 at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:4352)
 at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2933)
 at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10224)
 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1626)
 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1635)
 at android.view.Choreographer.doCallbacks(Choreographer.java:1170)
 at android.view.Choreographer.doFrame(Choreographer.java:1074)
 at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1599)
 at android.os.Handler.handleCallback(Handler.java:958)
 at android.os.Handler.dispatchMessage(Handler.java:99)
 at android.os.Looper.loopOnce(Looper.java:224)
 at android.os.Looper.loop(Looper.java:318)
 at android.app.ActivityThread.main(ActivityThread.java:8759)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:561)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
```

### RiMusic version

0.6.25

### Android version

Android 14

### Additional information

Crash1 looks related #982

**Comment by 821938089** [comment]
Crash2 related:
https://github.com/fast4x/RiMusic/blob/39fb2d0d7b01661d395dcc17551f57013d2015f1/app/src/main/kotlin/it/vfsfitvnm/vimusic/ui/screens/player/Queue.kt#L236
https://github.com/fast4x/RiMusic/blob/39fb2d0d7b01661d395dcc17551f57013d2015f1/app/src/main/kotlin/it/vfsfitvnm/vimusic/ui/screens/player/Queue.kt#L201
https://github.com/fast4x/RiMusic/blob/39fb2d0d7b01661d395dcc17551f57013d2015f1/app/src/main/kotlin/it/vfsfitvnm/vimusic/ui/screens/player/Queue.kt#L215-L216
https://github.com/fast4x/RiMusic/blob/39fb2d0d7b01661d395dcc17551f57013d2015f1/app/src/main/kotlin/it/vfsfitvnm/vimusic/ui/screens/player/Queue.kt#L221-L222
The  `-1` is the key point, I'm guessing it can be fixed by just changing it to `player.currentMediaItemIndex`.

**Comment by fast4x** [comment]
Hello I made the changes you indicated, thank you for the time you devoted and for the fix of the problem. I put your name in the project’s contributor developers. In the next release you will actually see if the problem resolves. Thanks again.

Available from version 0.6.26
